functions/zram.bash

```diff
@@ -1,63 +1,116 @@
+        if ! (whiptail --title "WARNING, Continue?" --yes-button "REALLY Continue" --no-button "Cancel" --yesno --defaultno "$lowMemText" 10 80); then echo "CANCELED"; return 0; fi
       fi
     fi
 
-    ZRamInstallLocation=/opt/zram
-    cond_redirect apt-get install -y -q --no-install-recommends make libattr1-dev
+    if ! dpkg -s 'make' 'libattr1-dev' &> /dev/null; then
+      echo -n "$(timestamp) [openHABian] Installing ZRAM required packages (make, libattr1-dev)... "
+      if cond_redirect apt-get install --yes make libattr1-dev; then echo "OK"; else echo "FAILED"; return 1; fi
+    fi
+
+    install_zram_code "$zramInstallLocation"
+
+    echo -n "$(timestamp) [openHABian] Setting up OverlayFS... "
+    if ! cond_redirect make --always-make --directory="$zramInstallLocation"/overlayfs-tools; then echo "FAILED (make overlayfs)"; return 1; fi
+    if ! mkdir -p /usr/local/lib/zram-config/; then echo "FAILED (create directory)"; return 1; fi
+    if cond_redirect install -m 755 "$zramInstallLocation"/overlayfs-tools/overlay /usr/local/lib/zram-config/overlay; then echo "OK"; else echo "FAILED (install overlayfs)"; return 1; fi
+
+    echo -n "$(timestamp) [openHABian] Setting up ZRAM... "
+    if ! install -m 755 "$zramInstallLocation"/openhabian-zram/zram-config /usr/local/bin/; then echo "FAILED (zram-config)"; return 1; fi
+    if ! cond_redirect install -m 644 "${BASEDIR:-/opt/openhabian}"/includes/ztab /etc/ztab; then echo "FAILED (ztab)"; return 1; fi
+    if ! mkdir -p /usr/local/share/zram-config/log; then echo "FAILED (create directory)"; return 1; fi
+    if ! cond_redirect install -m 644 "$zramInstallLocation"/openhabian-zram/ro-root.sh /usr/local/share/zram-config/ro-root.sh; then echo "FAILED (ro-root)"; return 1; fi
+    if cond_redirect install -m 644 "$zramInstallLocation"/openhabian-zram/zram-config.logrotate /etc/logrotate.d/zram-config; then echo "OK"; else echo "FAILED (logrotate)"; return 1; fi
 
-    install_zram_code "$ZRamInstallLocation"
-    cd "$ZRamInstallLocation"/overlayfs-tools || return 1
-    make
-    cd "$ZRamInstallLocation"/openhabian-zram || return 1
-    /bin/sh ./install.sh
-    /usr/bin/install -m 644 "${BASEDIR:=/opt/openhabian}"/includes/ztab /etc/ztab
+    if [[ -n $BATS_TEST_NAME ]]; then
+      sed -i 's|dir	lz4	150M		500M		/var/lib/openhab2/persistence	/persistence.bind||g' /etc/ztab
+    fi
+
+    echo -n "$(timestamp) [openHABian] Setting up ZRAM service... "
+    if ! cond_redirect install -m 644 "$zramInstallLocation"/openhabian-zram/zram-config.service /etc/systemd/system/zram-config.service; then echo "FAILED (copy service)"; return 1; fi
+    if ! cond_redirect systemctl enable zram-config.service; then echo "FAILED (enable service)"; return 1; fi
+    if cond_redirect systemctl restart zram-config.service; then echo "OK"; else echo "FAILED (restart service)"; return 1; fi
+  elif [[ $1 == "uninstall" ]]; then
+    echo -n "$(timestamp) [openHABian] Removing ZRAM service... "
```

Comment:
any particular reason for this hefty rework?

Did you test using menu? ZRAM is quite ....*sensitive* to changes e.g. in order

Reply:
Yes. Making it less dependent on upstream and making so we can better control it for our needs. It seems to work fine for me.